Il2cpp on mono - string tokens

brianradunity opened the pull request:
The new effort to run IL2CPP on the mono runtime instead of libil2cpp requires the metadata tokens for ldstr instructions, which Cecil currently discards.  This change stores these tokens in a dictionary on the MethodBody object, indexed by the instruction.

Commit:
Creating a new InlineString type that stores the inline string value as well as the metadata token, which we need to work with the mono runtime.

Commit:
Revert "Creating a new InlineString type that stores the inline string value as well as the metadata token, which we need to work with the mono runtime."

This reverts commit a6fe3492006f18f4dbf4d8782ac1b569edb0a24d.

Commit:
Storing the metadata token in a dictionary on the MethodBody instead of a different parameter type.  This will allow us to gain access to the token without breaking any code that is expecting a string parameter for a ldstr instruction.

mrvoorhe commented on Mono.Cecil.Cil/MethodBody.cs:
```diff
@@ -77,7 +79,7 @@ public sealed class MethodBody : IVariableDefinitionProvider {
 			set { scope = value; }
 		}
-
+		
```
Looks like an unnecessary spacing change.  If we are going to upstream this PR, they may appreciate removing it.

joncham replied:
Looks like extra whitespace?

joshpeterson reviewed (approved):
LGTM

joncham commented on Mono.Cecil.Cil/MethodBody.cs:
```diff
@@ -106,12 +108,25 @@ static ParameterDefinition CreateThisParameter (MethodDefinition method)
+
+		public bool GetInstructionToken (Instruction instruction, out MetadataToken token)
+		{
+			if (!instructionTokens.ContainsKey (instruction))
```
I think you could use 
`return instructionTokens .TryGetValue(instruction, token);`

mrvoorhe commented:
lgtm

joncham commented on Mono.Cecil.Cil/CodeReader.cs:
```diff
@@ -189,7 +189,9 @@ object ReadOperand (Instruction instruction)
-				return GetString (ReadToken ());
+				var token = ReadToken ();
+				body.instructionTokens.Add (instruction, token);
+				return GetString (token);
```
Does this handle all string literals?

TautvydasZilys commented:
Looks good to me.

Commit:
-Removing an unnecessary space.

-Replacing GetInstructionToken() body with a call to TryGetValue() on the instructionTokens dictionary.

joncham commented:
LGTM
